rustls/src/error.rs

```diff
@@ -466,22 +466,15 @@ pub enum CertificateError {
     BadSignature,
 
     /// A signature inside a certificate or on a handshake was made with an unsupported algorithm.
-    #[deprecated(
-        since = "0.23.29",
-        note = "use `UnsupportedSignatureAlgorithmContext` instead"
-    )]
-    UnsupportedSignatureAlgorithm,
-
-    /// A signature inside a certificate or on a handshake was made with an unsupported algorithm.
-    UnsupportedSignatureAlgorithmContext {
+    UnsupportedSignatureAlgorithm {
```

Comment:
I guess third-party validators can just leave these empty? Seems a bit ugly though.

Reply:
Do you want to retain no-context variants? Or wrap this in an `Option<Context>` thing?